src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -275,6 +285,11 @@ define(function (require) {
         return '' + self.toJSON();
       };
 
+      self.destroy = function () {
+        patternCache.clear(self.id);
```

Comment:
I don't think that `#destroy()` should require an id. Any index pattern should be destroyable